specutils/io/default_loaders/hst_cos.py

```diff
@@ -20,36 +20,45 @@ def cos_identify(origin, *args, **kwargs):
 
 
 @data_loader(label="HST/COS", identifier=cos_identify, extensions=['FITS', 'FIT', 'fits', 'fit'])
-def cos_spectrum_loader(file_name, **kwargs):
+def cos_spectrum_loader(file_obj, **kwargs):
     """
     Load COS spectral data from the MAST archive into a spectrum object.
 
     Parameters
     ----------
-    file_name: str
-        The path to the FITS file
+    file_obj: str, file-like, or HDUList
+         FITS file name, object (provided from name by Astropy I/O Registry),
+         or HDUList (as resulting from astropy.io.fits.open()).
 
     Returns
     -------
     data: Spectrum1D
         The spectrum that is represented by the data in this table.
     """
+    if isinstance(file_obj, fits.hdu.hdulist.HDUList):
+        close_hdulist = False
+        hdu = file_obj
+    else:
+        close_hdulist = True
+        hdu = fits.open(file_obj, **kwargs)
```

Comment:
Would there be serious overhead associated with just calling
`isinstance(file_obj, fits.hdu.hdulist.HDUList)`
again before the call to `hdu.close()`?
That would obviate the need for an extra variable.

Reply:
I think both of these are good suggestions, implemented.